apps/server/src/modules/lesson/service/lesson-copy.service.ts

```diff
@@ -266,29 +266,32 @@ export class LessonCopyService {
 	}
 
 	private copyLernStore(element: IComponentProperties): IComponentProperties {
-		const resources = ((element.content as IComponentLernstoreProperties).resources ?? []).map(
-			({ client, description, merlinReference, title, url }) => {
-				const result = {
-					client,
-					description,
-					merlinReference,
-					title,
-					url,
-				};
-				return result;
-			}
-		);
-
-		const lernstore = {
+		const lernstore: IComponentProperties = {
 			title: element.title,
 			hidden: element.hidden,
 			component: ComponentType.LERNSTORE,
 			user: element.user, // TODO should be params.user - but that made the server crash, but property is normally undefined
-			content: {
-				resources,
-			},
 		};
-		return lernstore as IComponentProperties;
+
+		if (element.content) {
+			const resources = ((element.content as IComponentLernstoreProperties).resources ?? []).map(
+				({ client, description, merlinReference, title, url }) => {
+					const result = {
+						client,
+						description,
+						merlinReference,
+						title,
+						url,
+					};
+					return result;
+				}
+			);
+
+			const lernstoreContent: IComponentLernstoreProperties = { resources };
```

Comment:
is this line needed? can't we just use lernstore.content = { resources }; ?

Reply:
@virgilchiriac indeed we could, but within that kind of complex structure that we have for that material stuff, I wanted to make clear (code wise) what exactly that `IComponentLernstoreProperties` determines, hence the extra line